XenAdmin/Wizards/PatchingWizard/PatchingWizard_PrecheckPage.cs

```diff
@@ -949,7 +955,28 @@ private void UpdateActionProgress(AsyncAction action)
 
         private void checkBox1_CheckedChanged(object sender, EventArgs e)
         {
-            RefreshRechecks();
+            RefreshDataGridViewList();
+        }
+
+        private void RefreshDataGridViewList()
+        {
+            dataGridView1.Rows.Clear();

```

Comment:
dataGridView1.Rows is used under a lock in other places (see the `AddRowToGridView` function). Is this lock really needed? If so, then we should also put this line inside a lock.